### PR TITLE
Add Feature Flags support

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Dear Reviewer

Given `prettier` was run here, feature-flags.js and test.js are the places to focus most of the energy on.

## What this does

1. Adds support for feature flags
2. Make everything `prettier`

## What this does not do

1. Refactor the existing codebase - I was originally doing this but decided to leave this up to a separate PR. Things like consolidating all the requests, etc.

## How the feature flags work

This follows the `posthog-python` implementation almost exactly. 

## Follow-up work

1. Documentation 
2. Writing up a feature flags spec - indeed the `posthog-python` code somewhat serves as this, but we need a place that at least tells us how to correctly calculate simple flags so we don't end up with diverging implementations. I spent some time playing around with `crypto` to get this right.
3. Refactor

### CODE REVIEW

File: feature-flags.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+const axios = require('axios')
+const crypto = require('crypto')
+const ms = require('ms')
+const version = require('./package.json').version
+
+const LONG_SCALE = 0xfffffffffffffff
+
+class ValueError extends Error {}
```
Comment (Twixes) [inline feature-flags.js:8]: I don't think it makes a ton of sense to simulate Python's `ValueError` in this JS code

File: index.d.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ declare module 'posthog-node' {
         host?: string
         api_host?: string
         enable?: boolean
+        personalApiKey?: string
+        featureFlagsPollingInterval?: number
```
Comment (Twixes) [inline index.d.ts:12]: Since existing options are `snake_case`, I think we should stick to that, even though it's a bit unusual for JS

**Review by Twixes — COMMENTED** [review]
_Please please_ never mix code formatting changes with behavior changes. I made #30 to make this PR reviewable. Now this PR only includes material updates to the code.
A couple of comments besides that but overall this looks reasonable.

File: feature-flags.js (hunk 1 above)
Comment (yakkomajuri) [reply feature-flags.js:8]: Yeah, I guess so - I had a few more of these around so they at least helped the dev, but I guess now it doesn't make sense, will remove

File: index.d.ts (hunk 2 above)
Comment (yakkomajuri) [reply index.d.ts:12]: Only `api_host` is - `flushAt` and `flushInterval` aren't. Happy to put another pull request to support `apiHost` and deprecate `api_host`. This lib is still small enough that we can make better decisions and not end up with the `posthog-js` behemoth

**Comment by yakkomajuri** [comment]
@Twixes yeah I do apologize for the `prettier` changes coming in this PR

**Comment by yakkomajuri** [comment]
Ready to go

File: index.d.ts (hunk 2 above)
Comment (yakkomajuri) [reply index.d.ts:12]: Actually, `api_host` hasn't been used for a while - it's only lingering here in the `d.ts` file, will remove it.

**Comment by yakkomajuri** [comment]
Also writing docs now

**Comment by weyert** [comment]
Awesome 🥇

**Pull request merged**